src/libDirectoryService/ViewChangePostProcessing.cpp

```diff
@@ -90,135 +93,127 @@ void DirectoryService::ProcessViewChangeConsensusWhenDone() {
-    // queue, waiting to be eject at the next ds epoch Suppose X1 - X3 are
-    // faulty ds nodes [X1][X2][X3][N1][N2] View change happen 3 times The view
-    // change will restructure the ds committee into this [X2][X3][N1][N2][X1]
-    // first vc. vc counter = 1 [X3][N1][N2][X1][X2] second vc. vc counter = 2
-    // [N1][N2][X1][X2][X3] third vc. vc counter = 3
-    // X1 m_consensusMyID = (size of ds committee) - 1 - faultyLeaderIndex
-    // (original)
-    //                    = 5 - 1 - 0 = 2
-    // X2 m_consensusMyID = (size of ds committee) - 1 - faultyLeaderIndex
-    // (original)
-    //                    = 5 - 1 - 1 = 3
-    // X3 m_consensusMyID = (size of ds committee) - 1 - faultyLeaderIndex
-    // (original)
-    //                    = 5 - 1 - 2 = 4
-
-    bool isCurrentNodeFaulty = false;
-    {
-      lock_guard<mutex> g2(m_mediator.m_mutexDSCommittee);
-
-      for (unsigned int faultyLeaderIndex = 0;
-           faultyLeaderIndex < m_viewChangeCounter; faultyLeaderIndex++) {
-        LOG_GENERAL(INFO,
-                    "Ejecting " << m_mediator.m_DSCommittee->front().second);
-
-        // Adjust ds commiteee
-        m_mediator.m_DSCommittee->push_back(m_mediator.m_DSCommittee->front());
-        m_mediator.m_DSCommittee->pop_front();
-
-        // Adjust faulty DS leader and/or faulty ds candidate leader
-        if (m_consensusMyID == faultyLeaderIndex) {
-          if (m_consensusMyID == 0) {
-            LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-                      "Current node is a faulty DS leader and got ousted "
-                      "by the DS "
-                      "Committee");
-          } else {
-            LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-                      "Current node is a faulty DS candidate leader and "
-                      "got ousted by the DS "
-                      "Committee");
-          }
-
-          // calculate (new) my consensus id for faulty ds nodes.
-          // Good ds nodes adjustment have already been done previously.
-          // m_consensusMyID = last index - num of time vc occur + faulty index
-          // Need to add +1 at the end as vc counter begin at 1.
-          m_consensusMyID = (m_mediator.m_DSCommittee->size() - 1) -
-                            m_viewChangeCounter + faultyLeaderIndex + 1;
-          isCurrentNodeFaulty = true;
-          LOG_GENERAL(INFO, "New m_consensusMyID  is " << m_consensusMyID);
-        }
+      // Remove faulty leader from the current ds committee structure temporary
+      if (iterFaultyLeader != m_mediator.m_DSCommittee->end()) {
+        m_mediator.m_DSCommittee->erase(iterFaultyLeader);
+      } else {
+        LOG_GENERAL(WARNING, "Cannot find "
+                                 << faultyLeader.second
+                                 << " to eject to back of ds committee");
+        // TODO: Handle this situation. This siutation shouldn't be
```

Comment:
Perhaps for scenarios like this we should start adding FATAL logs, so it's easier to discover in the testnet logs